tests_new/integration_tests/modules/datasets_base/test_dataset.py

```diff
@@ -6,39 +6,65 @@
     list_owned_datasets,
     list_datasets_created_in_environment,
 )
+from integration_tests.errors import GqlError
 
 log = logging.getLogger(__name__)
 
 
-def test_list_datasets(
+def test_list_datasets(  # TODO: add Redshift datasets and shared datasets
     client1, session_s3_dataset1, session_imported_sse_s3_dataset1, session_imported_kms_s3_dataset1, session_id
 ):
     assert_that(list_datasets(client1, term=session_id).nodes).is_length(3)
 
 
-def test_list_datasets_unauthorized(
+def test_list_datasets_unauthorized(  # TODO: add Redshift datasets and shared datasets
     client2, session_s3_dataset1, session_imported_sse_s3_dataset1, session_imported_kms_s3_dataset1, session_id
 ):
     assert_that(list_datasets(client2, term=session_id).nodes).is_length(0)
 
 
-def test_list_owned_datasets(  # TODO
+def test_list_owned_datasets(  # TODO: add Redshift datasets
     client1, session_s3_dataset1, session_imported_sse_s3_dataset1, session_imported_kms_s3_dataset1, session_id
 ):
     assert_that(list_owned_datasets(client1, term=session_id).nodes).is_length(3)
 
 
-def test_list_owned_datasets_unauthorized(  # TODO
+def test_list_owned_datasets_unauthorized(  # TODO: add Redshift datasets
     client2, session_s3_dataset1, session_imported_sse_s3_dataset1, session_imported_kms_s3_dataset1, session_id
 ):
     assert_that(list_owned_datasets(client2, term=session_id).nodes).is_length(0)
 
 
-def test_list_datasets_created_in_environment():
-    # TODO
-    pass
+def test_list_datasets_created_in_environment(  # TODO: add Redshift datasets
+    client1,
+    client2,
+    session_env1,
+    session_env2,
+    session_s3_dataset1,
+    session_imported_sse_s3_dataset1,
+    session_imported_kms_s3_dataset1,
+    session_id,
+):
+    assert_that(
+        list_datasets_created_in_environment(
+            client1, environment_uri=session_env1.environmentUri, term=session_id
+        ).nodes
+    ).is_length(3)
```

Comment:
The same thing we already discussed in previous PR and I don't know how do deal with it better.
But if we add more datasets, we will have to check all the tests like this, where the number is fixed. Is it ok?

Reply:
Yes, it is a case that could happen but it should not be the norm. Other modules should use the global_conftest fixtures from other modules, instead of creating their own. If they create their own, then they should create them in conftest and with a limited scope or without the term=session_id